src/lib.rs

```diff
@@ -38,45 +44,19 @@ impl TransClient {
         TransClient {
             url: url.to_string(),
             auth: None,
+            session_id: None,
+            client: Client::new(),
         }
     }
 
     /// Prepares a request for provided server and auth
     fn rpc_request(&self) -> reqwest::RequestBuilder {
-        let client = reqwest::Client::new();
         if let Some(auth) = &self.auth {
-            client
-                .post(&self.url)
+            self.client.post(&self.url)
                 .basic_auth(&auth.user, Some(&auth.password))
         } else {
-            client.post(&self.url)
-        }
-            .header(CONTENT_TYPE, "application/json")
-    }
-
-    /// Performs session-get call and takes the x-transmission-session-id
-    /// header to perform calls, using it's value
-    ///
-    /// # Errors
-    ///
-    /// If response is impossible to unwrap then it will return an empty session_id
-    async fn get_session_id(&self) -> String {
```

Comment:
Why did you remove that?

Reply:
In Spec they said that we should just send a req, then check for 409(6?) Http response, then obtain session id from there because it may change. Beside that, I think we should catch that session id for later use, do not need to get it on every req.